src/CMakeLists.txt

```diff
@@ -181,6 +181,30 @@ include(CheckCXXSymbolExists)
 # Finally, do the check!
 check_cxx_symbol_exists(BOOST_NO_CXX11_SMART_PTR boost/config.hpp HAVE_NO_CXX11_SMART_PTR)
 
+include(CheckCXXSourceCompiles)
+# Perform test to see if STIR unique_ptr exists
+check_cxx_source_compiles("
+   #include <memory>
+   std::unique_ptr<int> f(){int *ptr = new int;return std::unique_ptr<int>(ptr);}
+   int main(){auto p = f(); return 0;}
+   "
+        STIR_HAVE_UNIQUE_PTR)
+
+if(STIR_HAVE_UNIQUE_PTR)
+    option(STIR_NO_UNIQUE_PTR "Disable use of std::unique_ptr" OFF)
+else()
+    set(STIR_NO_UNIQUE_PTR ON)
+endif()
+
+# Perform test to see if BOOST shared_ptr exists
+check_cxx_source_compiles("
+        #include <memory>
+        std::shared_ptr<int> f(){int *ptr = new int; return std::shared_ptr<int>(ptr);}
+        int main(){auto p = f(); return 0;}
+    "
+        STIR_USE_BOOST_SHARED_PTR
+        )
+
 # Now we introduce 2 options such that the user can override the default behaviour
 # (not recommended! unless the above failed of course)
 option(STIR_NO_UNIQUE_PTR
```

Comment:
merge this with above code

Reply:
also need to fix STIR_USE_BOOST_SHARED_PTR option below in a similar way as the unique_ptr stuff above.

finally need to move the `mark_as_advanced` statement inside the `if`

Reply:
I am not sure if the mark_as_advanced stuff is implemented correctly.